### PR TITLE
distutils-r1.eclass: check that EPYTHON is set

### PR DESCRIPTION
**Pull request by Flowdalic**
Signed-off-by: Florian Schmaus <flow@gentoo.org>

### CODE REVIEW

File: eclass/distutils-r1.eclass [hunk 1]
```diff
@@ -820,6 +820,10 @@ distutils-r1_python_test() {
 		die "${FUNCNAME} can be only used after calling distutils_enable_tests"
 	fi
 
+	if [[ -n "${PYTHONPATH}" && "${PYTHONPATH@a}" != *x* ]]; then
```
Comment (mgorny) [inline eclass/distutils-r1.eclass:823]: Could you try checking if `EPYTHON` is set instead? That's the primary variable used to indicate we're in ebuild env.

Comment (Flowdalic) [reply eclass/distutils-r1.eclass]: good point. done

File: eclass/distutils-r1.eclass [hunk 2]
```diff
@@ -820,6 +820,10 @@ distutils-r1_python_test() {
 		die "${FUNCNAME} can be only used after calling distutils_enable_tests"
 	fi
 
+	if [[ -z ${EPYTHON} ]]; then
```
Comment (mgorny) [inline eclass/distutils-r1.eclass:823]: There are some other python* eclass functions that have similar check, please sync it to use the same syntax.

File: eclass/distutils-r1.eclass (hunk 1 above)
Comment (mgorny) [reply eclass/distutils-r1.eclass]: Cool. Could you cover other similar functions too? (i.e. `distutils-r1_python_compile` and so on)

File: eclass/distutils-r1.eclass (hunk 2 above)
Comment (Flowdalic) [reply eclass/distutils-r1.eclass:823]: Yep, I've saw those, but I decided to mimic the `if not-set; then die` pattern that is used in the function. Which is also the pattern I prefer. But I can change it back to `[[ -n ${EPYTHON} ]] || die "EPYTHON unset, invalid call context"`.

**Pull request closed without merging**